### ISSUE TITLE
Deleting a Project Leaves Orphaned Comments

### ISSUE THREAD

**Issue by rickbau5** [uidaho/squireproject#114]
If you delete a project, the comments for that project are not deleted. This could be handled manually in code. But the more robust way would be with Foreign Keys and making our tables cascading. [Laravel doc on it](https://laravel.com/docs/4.2/schema#foreign-keys). 

This can be expanded to other tables as well. For instance, the upcoming table in #111 shows if a user is following a project. If the User is deleted, the corresponding `ProjectFollower` entry should be deleted. Furthermore, if a Project is deleted, it should also be updated.

**Comment by rickbau5** [comment]
Also a simpler way to do this would be to customize the delete method for the Project model further. I already did to delete images, could extend that to delete comments as well.

**Comment by brandonjank** [comment]
I experimented with adding foreign keys using the laravel migration process last night. Its not worth the mess. The easiest thing to get er done before end of semester is for the project controller to explicitly drop all rows corresponding to the project_id.

**Comment by rickbau5** [comment]
yeah makes sense. If you call `delete` on a model, then we can override that method to delete corresponding stuff as well. Like so:

``` php
    public function delete()
    {
        $res = parent::delete();
        $imagePath = base_path() . '/public' .$this->getImagePath();
        if (file_exists($imagePath)) {  // Shouldn't be null, let's check for sanity
            unlink($imagePath);
        }
        return $res;
    }
```

**Comment by brec9824** [comment]
Added deleting comments and everything connected to the project in pull #147

**Comment by brec9824** [comment]
Fixed in pull #147
